homeassistant/components/nissan_leaf/binary_sensor.py

```diff
@@ -3,27 +3,26 @@
 
 from homeassistant.components.nissan_leaf import (
     DATA_LEAF, DATA_PLUGGED_IN, LeafEntity)
+from homeassistant.components.binary_sensor import BinarySensorDevice
 
 _LOGGER = logging.getLogger(__name__)
 
 DEPENDENCIES = ['nissan_leaf']
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
+def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up of a Nissan Leaf binary sensor."""
-    _LOGGER.debug(
-        "binary_sensor setup_platform, discovery_info=%s", discovery_info)
-
     devices = []
```

Comment:
Add a guard clause here that checks if discovery_info is None and return if so. That should be first in the function body.

Reply:
Added guard clauses to binary_sensor, sensor and switch in commit https://github.com/home-assistant/home-assistant/pull/21145/commits/27d28b0f10cbe70f5f25a3d0b6405dc951b5cc42